apps/web/vibes/soul/sections/products-list-section/sorting.tsx

```diff
@@ -14,50 +14,62 @@ export interface Option {
 }
 
 export function Sorting({
-  label,
-  options,
+  label: streamableLabel,
+  options: streamableOptions,
   paramName = 'sort',
   defaultValue = '',
+  placeholder: streamablePlaceholder,
 }: {
   label?: Streamable<string | null>;
   options: Streamable<Option[]>;
   paramName?: string;
   defaultValue?: string;
+  placeholder?: Streamable<string | null>;
 }) {
   return (
-    <Suspense fallback={<SortingSkeleton />}>
-      <SortingInner
-        defaultValue={defaultValue}
-        label={label}
-        options={options}
-        paramName={paramName}
-      />
-    </Suspense>
+    <Stream
+      fallback={<SortingSkeleton />}
+      value={Promise.all([streamableLabel, streamableOptions, streamablePlaceholder])}
```

Comment:
We cannot use `Promise.all` in a client component because that will make a new promise on render which will cause React to throw an error. Did this not throw when testing locally with the examples?

Reply:
No errors on my end, but I moved the the `Stream` component inside into `ProductListSection` so I could use `Promise.all()`.